security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -646,7 +662,14 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 				// If token is still valid, re-generated the secret and push change to proxy.
 				// Most likey this code path may not necessary, since TTL of cert is much longer than token.
 				// When cert has expired, we could make it simple by assuming token has already expired.
-				ns, err := sc.generateSecret(context.Background(), secret.Token, connKey, now)
+				var ns *model.SecretItem
+				var err error
+				if SupportRotationWithCert {
```

Comment:
Simpler if you call the function in once place:

```go
token := EmptyToken
if SupportRotationWithCert {
  token = secret.Token
}
ns, err := sc.generateSecret(context.Background(), token, connKey, now, !SupportRotationWithCert)
```

Reply:
Yes that can make code less. But I think will make the readability a little less. As you mixed the token and RotationWithCert flag together with !SupportRotationWithCert in a function. That will takes time to think the logic: 1. with cert, no token 2. without cert, with token

Reply:
Let's step back a little bit, why do we need to pass EmptyToken if withToken is already set to false? I thought the token is just ignored if withToken is set to false?

Reply:
Yes exactly. in the CSR sign if the withtoken is set to false. It will not append the token into its context

Reply:
if so why not just pass the `secret.Token` as is? It's not going to be used anyway if `withToken` is false. Maybe rename the `withToken` to an enum that has two values `USE_TOKEN` and `USE_CERTIFICATE`.

Reply:
rewrite it